.github/workflows/ci.yaml

```diff
@@ -8,40 +8,43 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: contracts-periphery/
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
 
+      - uses: volta-cli/action@v1
+
       - name: Install Foundry
         uses: onbjerg/foundry-toolchain@v1
         with:
           version: nightly
 
-      - name: Build contracts
-        run: |
-          forge --version
-          forge build --sizes
+      - run: yarn install --network-concurrency 1
+
+      - run: yarn build
   tests:
     runs-on: ubuntu-latest
     env:
       FOUNDRY_PROFILE: ci
       ETH_RPC_URL: ${{ secrets.ETH_RPC_URL }}
-    defaults:
-      run:
-        working-directory: contracts-periphery/
+      INFURA_ID: ${{ secrets.INFURA_ID }}
+      ETHERSCAN_API_KEY: ${{ secrets.ETHERSCAN_API_KEY }}
+      POLYGONSCAN_API_KEY: ${{ secrets.POLYGONSCAN_API_KEY }}
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
 
+      - uses: volta-cli/action@v1
```

Comment:
I notice both build and test install foundry, volta, and yarn packages. The yarn install step takes 5 min on each job! I looked a little into whether portions of the workflow can be reused across jobs, but it doesn't seem like it

Reply:
Thanks for looking into this. I couldn't find a way either. It's a bummer we duplicate this but I don't see a way to change it.